### DISCUSSION TITLE
What is required to replace the default base maps in TerriaMap?

### DISCUSSION THREAD

**Discussion by lstillwe** [TerriaJS/terriajs#5068] [Q&A] [answered]
What is required to replace the default base maps in TerriaMap? - i.e. different sources, thumbnails, etc.
Do I need to create new ViewModel?

Thanks!

**Comment by hardreddata** [comment, marked as answer]
Hi @lstillwe 

Yes, you can do this in TerriaMap.

I copied the whole `createGlobalViewModels.js` from TerriaJS into my TerriaMap fork `lib/ViewModels` and modified it by adding a new map.

Then edit `index.js` such that you use your modified copy rather than the one from TerriaJS 

`var createGlobalBaseMapOptions = require('./lib/ViewModels/createGlobalBaseMapOptions');`

I think this is the recommended approach. Someone else will tell us if there is a better way. 

Good luck.
